### PR TITLE
Add FXIOS-8540 | Add new getLoginsForProtectionSpace on RustLogins to remove Deferred

### PR DESCRIPTION
**Pull request by imsanchit**
## :scroll: Tickets
[Jira ticket](https://mozilla-hub.atlassian.net/browse/FXIOS-8540)
[Github issue](https://github.com/mozilla-mobile/firefox-ios/issues/18961)

## :pencil: Checklist
You have to check all boxes before merging
- [x] Filled in the above information (tickets numbers and description of your work)
- [x] Updated the PR name to follow our [PR naming guidelines](https://github.com/mozilla-mobile/firefox-ios/wiki/Pull-Request-Naming-Guide)
- [ ] Wrote unit tests and/or ensured the tests suite is passing
- [ ] When working on UI, I checked and implemented accessibility (minimum Dynamic Text and VoiceOver)
- [ ] If needed, I updated documentation / comments for complex code and public methods
- [ ] If needed, added a backport comment (example `@Mergifyio backport release/v120`)

### CODE REVIEW

File: firefox-ios/Storage/Rust/RustLogins.swift [hunk 1]
```diff
@@ -688,6 +720,23 @@ public class RustLogins: LoginsProtocol {
         }
     }
 
+    public func listLogins(completionHandler: @escaping (Result<[EncryptedLogin], Error>) -> Void) {
```
Comment (lmarceau) [inline firefox-ios/Storage/Rust/RustLogins.swift:723]: Doesn't this one duplicate the function you added inhttps://github.com/mozilla-mobile/firefox-ios/pull/19206?

Comment (imsanchit) [reply firefox-ios/Storage/Rust/RustLogins.swift:723]: yes it looks like duplicate stuff, probably the merge order would help here

Comment (lmarceau) [reply firefox-ios/Storage/Rust/RustLogins.swift:724]: Mh. I tried fixing the conflicts in the Github page directly. Seems this is still showing up but it shouldn't as the other PR was merged 🤔 Can you fix this @imsanchit?

Comment (imsanchit) [reply firefox-ios/Storage/Rust/RustLogins.swift:727]: @lmarceau please check if this PR can be merged now.

Comment (cyndichin) [reply firefox-ios/Storage/Rust/RustLogins.swift:727]: @imsanchit It seems that there is a another similar function from the other PR. Should we remove that one to and replace it with this one? 

Line: 765

image

**Comment by cyndichin** [comment]
@imsanchit let me know if this PR is ready for another review, just making sure you're not waiting on me! Seems there's some conflicts and references to Deferred, so might be in progress.

**Comment by imsanchit** [comment]
Thanks for the heads up @cyndichin, I have updated the PR and fixed the conflict and deferred usage. Please have a look at it now.

**Review by cyndichin — APPROVED** [review]
thanks for the update @imsanchit! looks great :)

**Pull request merged**